src/script/lua_api/l_object.cpp

```diff
@@ -1806,6 +1806,36 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+int ObjectRef::l_set_camera_mode_restriction_flags(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = get_player(ref);
+	if (!player)
+		return 0;
+
+
+}
+
+int ObjectRef::l_get_camera_mode_restriction_flags(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = get_player(ref);
+	if (!player)
+		return 0;
+
+	lua_newtable(L);
+	lua_pushboolean(L, player->camera_mode_restrictions & 0x01);
```

Comment:
use the getter instead of the raw attribute please

Reply:
I think the getter/setter methods are only accessible by the clients, and not servers - are files in `src/client/` compiled when building the server?

Reply:
Nvm, added getter/setter methods for `RemotePlayer::m_camera_mode_restrictions`